tests/jdom/csvtrace.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { suite, test } from "mocha"
+import { ButtonEvent } from "../../src/jdom/constants"
+import {
+    withTestBus,
+    nextEventFrom,
+    nextUpdateFrom,
+    createServices,
+} from "./tester"
+import { FastForwardScheduler } from "./scheduler"
+import { assert } from "../../src/jdom/utils"
+import {
+    SRV_POTENTIOMETER,
+    SystemReg,
+} from "../../src/jdom/constants"
+import SensorServer from "../../src/servers/sensorserver"
+import ButtonServer from "../../src/servers/buttonserver"
+import { SensorServerCsvSource } from "./sensorcsvsource"
+
+// Configured to ignore differences for a 16-bit fixed point
+function approxEquals(a: number, b: number, maxPctDiff = 1 / 32767) {
+    return Math.abs(b - a) / Math.min(b, a) < maxPctDiff
+}
+
+suite('"CSV" trace server', () => {
+    test("produces register data as expected", async function () {
+        await withTestBus(async bus => {
+            const { pot } = await createServices(bus, {
+                pot: new SensorServer<[number]>(SRV_POTENTIOMETER),
+            })
+            const potStreamer = new SensorServerCsvSource(pot.server, [
+                // takes about 500ms for the bus to spin up
```

Comment:
See csv.proxy for the CSV parser in jacdac docs. We typically use object where keys = csv headers rather than arrays.

Reply:
Done (I think? I didn't get the csv.proxy running on locally but it seems to use Papa Parse, so I made the input consistent with that).